pkg/storage/unified/README.md

```diff
@@ -781,3 +781,618 @@ The dual writer system provides metrics for monitoring:
 - `dual_writer_mode_transitions_total`: Counter of mode transitions
 
 Use these metrics to monitor the health of your migration and identify any issues with the dual writer system.
+
+---
+
+## Unified Search System
+
+The Unified Search system provides a scalable, distributed search capability for Grafana's Unified Storage. It uses a ring-based architecture to distribute search requests across multiple search server instances, with namespace-based sharding for optimal performance and data distribution.
+
+### System Architecture
+
+The search system provides both unified and legacy search capabilities, with routing based on dual writer mode configuration:
+
+```mermaid
+graph TB
+    subgraph "Request Sources"
+        A[Grafana UI<br/>User Search]
+        B[Dashboard Service<br/>Resource Searches]
+        C[Folder Service<br/>Resource Searches]
+        D[Alerting Service<br/>Resource Searches]
+        E[Provisioning Service<br/>Resource Searches]
+        F[API Endpoints<br/>Search Operations]
+    end
+    
+    subgraph "API Gateway Layer"
+        G[Grafana API Server<br/>Search Endpoint]
+        H[Search Client<br/>Dual Writer Aware]
+    end
+    
+    subgraph "Routing Decision"
+        I{Dual Writer Mode<br/>Check}
+    end
+    
+    subgraph "Unified Search Path (Mode 3+)"
+        J[Search Distributor<br/>Ring-based Routing]
+        K[Ring<br/>Consistent Hashing]
+        L[Search API Server 1<br/>Namespace Sharding<br/>+ Embedded Bleve Backend]
+        M[Search API Server 2<br/>Namespace Sharding<br/>+ Embedded Bleve Backend]
+        N[Search API Server 3<br/>Namespace Sharding<br/>+ Embedded Bleve Backend]
+        O[Unified Storage<br/>K8s-style Resources]
+    end
+    
+    subgraph "Legacy Search Path (Mode 0-2)"
+        Q[Legacy Search Service<br/>SQL-based Search]
+        R[Legacy Storage<br/>Traditional Tables]
+        S[Shadow Traffic<br/>Mode 1-2 + Flag Enabled]
+    end
+    
+    A --> G
+    B --> H
+    C --> H
+    D --> H
+    E --> H
+    F --> G
+    G --> H
+    H --> I
+    
+    I -->|Mode 3+| J
+    I -->|Mode 0-2| Q
+    
+    J --> K
+    K --> L
+    K --> M
+    K --> N
+    L -.->|Just-in-Time<br/>Indexing| O
+    M -.->|Just-in-Time<br/>Indexing| O
+    N -.->|Just-in-Time<br/>Indexing| O
+    
+    Q --> R
+    Q -.->|Shadow Traffic<br/>Mode 1-2 + Flag| S
+    S -.-> J
+```
+
+### Search Backend Routing
+
+The search client routes requests based on the dual writer mode configuration for each resource type:
+
+#### Dual Writer Mode → Backend Routing
+- **Mode 0-2**: Route to **Legacy Search**
+  - Mode 1-2: Shadow traffic to Unified Search (if `unifiedStorageSearchDualReaderEnabled` is enabled)
+  - Mode 0: No shadow traffic
+- **Mode 3+**: Route to **Unified Search** 
+  - No shadow traffic needed (unified is primary)
+
+### Feature Flags
+
+Unified Search requires several feature flags to be enabled depending on the desired functionality:
```

Comment:
It might be helpful mentioning the feature flags which are prerequisites (unified storage and k8s app platform ones) or linking to documentation about them.